test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/MSBuildIntegrationTestBase.cs

```diff
@@ -46,5 +48,29 @@ internal Task<MSBuildResult> DotnetMSBuild(string target, string args = null, bo
 
             return MSBuildProcessManager.RunProcessAsync(Project, $"{restoreArgument} /t:{target} /p:Configuration={Configuration} {args}", timeout);
         }
+
+        /// <summary>
+        /// Locks all files, discovered at the time of method invocation, under the
+        /// specified <paramref name="directory" /> from reads or writes.
+        /// </summary>
+        public IDisposable LockDirectory(string directory)
+        {
+            var disposables = new List<IDisposable>();
+            foreach (var file in Directory.EnumerateFiles(directory, "*", SearchOption.AllDirectories))
```

Comment:
It would be nice to do a `Path.Combine(ProjectDirectory.DirectoryPath, directory)` here - this is the convention that I've been doing in other cases. This means that it will work in both cases (if `directory` is rooted or relative to the project).